src/api/auth/types.ts

```diff
@@ -0,0 +1,30 @@
+export type SignupResponse = {
+  status: 'succes' | 'error';
+  data: Data;
+  errors?: Errors;
+};
+
+export type SigninRespose = {
+  data: Data;
+};
+
+type Data = {
+  email: string;
+  provider: string;
+  uid: string;
+  id: number | null;
+  allow_password_change: boolean;
+  name: string;
+  nickname: any | null;
```

Comment:
In most projects, you might face that the use of `any` is prohibited or most likely to be avoided.

This is a non-blocking suggestion.

```suggestion
  nickname: string | null;
```